packages/hub/services/scheduled-payments/fetcher.ts

```diff
@@ -19,44 +21,74 @@ export default class ScheduledPaymentsFetcherService {
-                },
-              },
-            },
-          },
-          {
-            AND: {
-              recurringDayOfMonth: {
-                gt: 0,
-              },
-              recurringUntil: {
-                gte: nowUtc,
-              },
-              scheduledPaymentAttempts: {
-                none: {
-                  status: 'inProgress',
-                },
-              },
-            },
-          },
-        ],
-      },
-    });
+    let results: [{ id: string; started_at: Date }] = await prisma.$queryRaw`SELECT 
+        "scheduled_payments"."id" AS "id",
+        "last_failed_payment_attempts"."started_at"
+      FROM 
+        "scheduled_payments"
+      LEFT JOIN (SELECT "scheduled_payment_attempts"."scheduled_payment_id",
+                  MAX("scheduled_payment_attempts"."started_at") AS "started_at"
+            FROM "scheduled_payment_attempts"
+            WHERE "scheduled_payment_attempts"."status" = 'failed'
+            GROUP BY "scheduled_payment_id") AS "last_failed_payment_attempts"
+      ON "last_failed_payment_attempts"."scheduled_payment_id" = "scheduled_payments"."id"
+      WHERE 
+        (
+          "scheduled_payments"."canceled_at" IS NULL 
+          AND "scheduled_payments"."creation_block_number" > 0 
+          AND "scheduled_payments"."pay_at" > ${startOfDay(subDays(nowUtc, this.validForDays))}
+          AND "scheduled_payments"."pay_at" <= ${nowUtc}
+          AND (
+            (
+              "scheduled_payments"."recurring_day_of_month" IS NULL 
+              AND (
+                "scheduled_payments"."id"
+              ) NOT IN (
+                SELECT 
+                  "t0"."id" 
+                FROM 
+                  "scheduled_payments" AS "t0" 
+                  INNER JOIN "scheduled_payment_attempts" AS "j0" ON ("j0"."scheduled_payment_id") = ("t0"."id") 
+                WHERE 
+                  (
+                    (
+                      "j0"."status" = 'succeeded'
+                      OR "j0"."status" = 'in_progress'
+                    ) 
+                    AND "t0"."id" IS NOT NULL
+                  )
+              )
+            ) 
+            OR (
+              "scheduled_payments"."recurring_day_of_month" > 0 
+              AND "scheduled_payments"."recurring_until" >= ${nowUtc}
+              AND (
+                "scheduled_payments"."id"
+              ) NOT IN (
+                SELECT 
+                  "t0"."id" 
+                FROM 
+                  "scheduled_payments" AS "t0" 
+                  INNER JOIN "scheduled_payment_attempts" AS "j0" ON ("j0"."scheduled_payment_id") = ("t0"."id") 
+                WHERE 
+                  (
+                    "j0"."status" = 'in_progress'
+                    AND "t0"."id" IS NOT NULL
+                  )
+              )
+            )
+          )
+        )
+        ORDER BY CASE WHEN  "last_failed_payment_attempts"."started_at" IS NULL THEN 0 ELSE 1 END
```

Comment:
Did you mean `ORDER BY CASE WHEN  last_failed_payment_attempts.started_at IS NULL THEN 0 ELSE started_at END`?

Reply:
No, it will not works because started_at is a `timestamp without timezone` we need an integer on that `ELSE` scope.

Reply:
Updated, make it more readable `ORDER BY CASE WHEN last_failed_payment_attempt.started_at IS NULL THEN ${convertDateToUTC(new Date(0))} ELSE last_failed_payment_attempt.started_at END`